src/csharp/Microsoft.Spark.UnitTest/CallbackTests.cs

```diff
@@ -153,17 +153,33 @@ public void TestCallbackHandlers()
             clientSocket.Connect(ipEndpoint.Address, ipEndpoint.Port);
 
             var callbackConnection = new CallbackConnection(0, clientSocket, callbackHandlersDict);
-            Task.Run(() => callbackConnection.Run(token));
+            Task task = Task.Run(() => callbackConnection.Run(token));
 
-            using ISocketWrapper serverSocket = serverListener.Accept();
-            WriteAndReadTestData(serverSocket, callbackHandler, inputToHandler, token);
+            if (token.IsCancellationRequested)
```

Comment:
How have we verified this change removes the flakiness of these tests? Do we need to run this multiple times to confirm?

Reply:
Yes, this has already been verified.